### ISSUE TITLE
PHP 8.1

### ISSUE THREAD

**Issue by olape-git** [cmsimple-xh/cmsimple-xh#549]
Since PHP 7.4 will be phased out in November 2022, we should introduce support for PHP 8.1 in this release.

**Comment by olape-git** [comment]
But maybe it's not so bad / much.
Then this would definitely be an option for the 1.7.6.
But I have not tested it myself yet.

**Comment by cmb69** [comment]
> Does this mean that I should not test with PHP 8.1 at all (xampp PHP 8.1.1)?

No, on the contrary. I was asking, whether some problems have already been found. :)

**Comment by cmb69** [comment]
> But maybe it's not so bad / much.

Maybe, maybe not. I expect at least deprecations due to https://wiki.php.net/rfc/deprecate_null_to_scalar_internal_arg.

**Comment by frase-git** [comment]
XH 1.7.5 unmodified original version
PHP 8.1.1 (xampp WIN)

Debugmode wrote:

DEPRECATED: Return type of XH\PluginConfig::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
... \cmsimple\classes\PluginConfig.php:52
DEPRECATED: Return type of XH\PluginConfig::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
... \classes\PluginConfig.php:67
DEPRECATED: Return type of XH\PluginConfig::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
... \classes\PluginConfig.php:83
DEPRECATED: Return type of XH\PluginConfig::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
... \classes\PluginConfig.php:98

**Comment by cmb69** [comment]
@frase-git, ah right, fixed: f63148976ddd07d49ff2afd928f7b0a6e70ee97f (this nicely shows that we need to fade out support for old PHP versions)

**Comment by cmb69** [comment]
Mal was zum Aufmuntern für Zwischendurch:

![Screenshot 2023-01-20 195856](https://user-images.githubusercontent.com/2306138/213783162-73870507-3156-4897-b5ac-2042ae8d00e7.gif)

PHP 8.2 nehme ich mir lokal vor; wird wohl in der CI nicht praktikabel sein.
